OfficeDevPnP.Core/OfficeDevPnP.Core/Framework/Provisioning/ObjectHandlers/ObjectPages.cs

```diff
@@ -69,6 +69,12 @@ public override void ProvisionObjects(Web web, ProvisioningTemplate template)
                     web.AddWikiPageByUrl(url);
                     web.AddLayoutToWikiPage(page.Layout, url);
                 }
+
+                if (page.WelcomePage)
+                {
+                   web.SetHomePageByServerRelativePath(url);
```

Comment:
I understand your reasoning. I think though that we shouldn't add new extension methods because one specific scenario requires it. It shouldn't be too complex to create a rootfolder relative url from the server relative url in the object handler and then use that one instead?